src/wakeup.c

```diff
@@ -1,67 +1,110 @@
 #include <appling.h>
 #include <assert.h>
+#include <path.h>
 #include <stddef.h>
 #include <stdlib.h>
 #include <string.h>
 #include <uv.h>
 
 #include "../include/wakeup.h"
-#include "platform.h"
 
-static appling_app_t *wakeup_apps = NULL;
-static size_t wakeup_apps_len = 0;
+typedef struct {
+  char *url;
+  appling_platform_t platform;
 
-static uv_once_t wakeup_init_guard = UV_ONCE_INIT;
+  struct {
+    uv_process_t process;
+    appling_resolve_t resolve;
+    appling_lock_t lock;
+  } handles;
+} wakeup_t;
 
 static void
-on_paths (appling_paths_t *req, int status, const appling_app_t *apps, size_t apps_len) {
-  assert(status == 0);
-
-  wakeup_apps = malloc(apps_len * sizeof(appling_app_t));
-  wakeup_apps_len = apps_len;
+on_process_exit (uv_process_t *handle, int64_t exit_status, int term_signal) {
+  uv_close((uv_handle_t *) handle, NULL);
+}
 
-  memcpy(wakeup_apps, apps, apps_len * sizeof(appling_app_t));
+static void
+on_unlock (appling_lock_t *req, int status) {
+  assert(status == 0);
 }
 
 static void
-on_wakeup_init (void) {
+on_resolve (appling_resolve_t *req, int status) {
   int err;
 
-  uv_loop_t loop;
-  err = uv_loop_init(&loop);
+  assert(status == 0);
+
+  wakeup_t *wakeup = (wakeup_t *) req->data;
+
+  appling_path_t pear;
+  size_t path_len = sizeof(appling_path_t);
+
+  err = path_join(
+    (const char *[]){req->platform->path, "bin", "pear-runtime", NULL},
```

Comment:
is this pointing to the by-arch folder or the bin symlink in platform dir?

Reply:
It points to `by-arch/<platform>-<arch>/bin/pear-runtime`.

Reply:
Then all good